### PR TITLE
Add support for parsing out unit number and adding to results

### PR DESCRIPTION
**Pull request by gauthierm**
1. Parses unit information out of original query string
2. Passes string without unit information to Google Places
3. re interpolates unit information back into places autocomplete results
4. Includes unit information in submitted autocomplete object

### CODE REVIEW

File: src/parseUnitNumber.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+// Mapping of full unit designator string to preferred abbreviation.
+const normalizedDesignators = {
+  '#': '',
+  apartment: 'apt',
+  building: 'bldg',
+  department: 'dept',
+  floor: 'fl',
+  room: 'rm',
+  suite: 'ste',
+  unit: 'unit'
```
Comment (nburka) [inline src/parseUnitNumber.ts:26]: what about the french versions of these? Canada Post lists some of them here https://www.canadapost.ca/tools/pg/manual/PGaddress-e.asp?ecid=murl10006450#1442083

**Review by nburka — COMMENTED** [review]
What about the other variations that we came up with here? https://docs.google.com/document/d/1X2P9_jJvzl8Z2lvO6Mlk1V5PG5D9lAK0dCGUVZ44SXA/edit

See the USPS and Canada Post links in that file.

File: src/parseUnitNumber.ts [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+      unitDesignator: '',
+      unitNumber: dashedLetterMatches[2].toUpperCase()
+    };
+  }
+
+  // Match unit number and designator after the civic number and before the
+  // street name. Needs to go before after-street case to prevent parsing civic
+  // number as a full street address.
+  const afterNumber = new RegExp(
+    `^\\s*([0-9]+)\\s+(${designators})\\s*([0-9]+|[a-z])([\\s,].*)$`,
```
Comment (nburka) [inline src/parseUnitNumber.ts]: can we optionally also match letters after numbers? Like `2A` or `5C`? Often apartment numbers are done that way.

**Review by nburka — CHANGES_REQUESTED** [review]

**Comment by nburka** [comment]
When you re-interpolate it back into the result, it would read better to me to have a comma before the unit info, like `3759 Rue Saint-André, apt 2, Montreal, QC, Canada` vs. `3759 Rue Saint-André apt 2, Montreal, QC, Canada` like it is now.

Unless this is a standard you're following that I'm not aware of.

**Comment by gauthierm** [comment]
> When you re-interpolate it back into the result, it would read better to me to have a comma before the unit info.

Canada Post and USPS exclude a comma in that case.

File: src/parseUnitNumber.ts (hunk 2 above)
Comment (gauthierm) [reply src/parseUnitNumber.ts]: Like `Apartment 1I`?

**Pull request merged**